pubsub/consumer.go

```diff
@@ -49,8 +51,9 @@ type Consumer[Request any, Response any] struct {
 }
 
 type Message[Request any] struct {
-	ID    string
-	Value Request
+	ID          string
+	Value       Request
+	AckNotifier chan struct{}
```

Comment:
now, let's make ackNotifier a private field, and add an Ack() function that closes it.